service/src/main/java/com/bankbazaar/kafka/service/service/FileStatusService.java

```diff
@@ -36,11 +33,17 @@ public FileStatusEntity createEntry(DataDto data)
         FileStatusEntity response = insert(fileData);
         return response;
     }
-    public void updateEntry(Data data, Status status)
+    public FileStatusEntity getEntry(Long id)
+    {
+        return fileStatusManager.getEntry(id).get();
+    }
+    public FileStatusEntity updateEntry(Data data, Status status)
```

Comment:
Take only the dataId and status as input.
The function should be like
1. Find entity by Id
2. Set status
3. Save

The main idea behind this is since there are multiple functions which want to update the status why do all the functions send entire objects? Only the id and new status should be enough.

Also when I know I am only going to update the status why do I need a full update function?

Reply:
Update entry has been modified to take only id and status as arguments.